thavalon-server/src/database/accounts/mod.rs

```diff
@@ -31,19 +33,25 @@ struct InternalDBAccount {
     email_verified: bool,
 }
 
-impl From<DatabaseAccount> for InternalDBAccount {
-    fn from(public_account: DatabaseAccount) -> Self {
-        InternalDBAccount {
-            _id: ObjectId::with_string(&public_account.id).expect(&format!(
-                "Received ID {} is not a valid ObjectId.",
-                &public_account.id
-            )),
+impl TryFrom<DatabaseAccount> for InternalDBAccount {
+    type Error = String;
+
+    fn try_from(public_account: DatabaseAccount) -> Result<Self, Self::Error> {
+        let _id = match ObjectId::with_string(&public_account.id) {
+            Ok(id) => id,
+            Err(_) => {
```

Comment:
Is it possible to log a stack trace or something? Or does rust automatically do that? Just curious why not printing anything about actual error here.

Reply:
Fixed